nixos/modules/services/networking/gns3.nix

```diff
@@ -0,0 +1,261 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.gns3;
+
+  configFile = pkgs.writeText "gns3-server.conf" ''
+    [Server]
+    appliances_path = /var/lib/gns3/appliances
+    configs_path = /var/lib/gns3/configs
+    images_path = /var/lib/gns3/images
+    projects_path = /var/lib/gns3/projects
+    symbols_path = /var/lib/gns3/symbols
+
+    ${lib.optionalString (cfg.ubridge.enable) ''
+    ubridge_path = ${cfg.ubridge.package}/bin/ubridge
+    ''}
+
+    auth = ${if cfg.auth.enable then "True" else "False"}
+    ${lib.optionalString (cfg.auth.user != null) ''
+      user = ${cfg.auth.user}
+    ''}
+    ${lib.optionalString (cfg.auth.password != null) ''
+      password = ${cfg.auth.password}
+    ''}
+
+    ${lib.optionalString (cfg.vpcs.enable) ''
+    [VPCS]
+    vpcs_path = ${cfg.vpcs.package}/bin/vpcs
+    ''}
+
+    ${lib.optionalString (cfg.dynamips.enable) ''
+    [Dynamips]
+    dynamips_path = ${cfg.dynamips.package}/bin/dynamips
+    ''}
+  '';
+
+in {
+  meta.maintainers = [ lib.maintainers.anthonyroussel ];
+
+  options = {
+    services.gns3 = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc ''Whether to enable GNS3 Server daemon.'';
+      };
+
+      package = lib.mkPackageOptionMD pkgs "gns3-server" {};
+
+      auth = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = false;
+          description = lib.mdDoc ''Enable password based HTTP authentication to access the GNS3 server.'';
+        };
+
+        user = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          default = null;
+          example = "gns3";
+          description = lib.mdDoc ''Username used to access the GNS3 server.'';
+        };
+
+        password = lib.mkOption {
```

Comment:
Offering a passwordFile alternative is probably a 'must' here.

Reply:
At the moment, I have no idea how to do this properly.

GNS3 Server doesn't provide any environment or configuration variables to set the path to a secret password file (to avoid storing it into the Nix store).

Maybe I can try to do that using a preStart entry in the SystemD unit, and modifying the GNS3 server configuration file dynamically with the password when the service starts.

I'm looking into how to do it this weekend.

Reply:
Fixed, used replace-secret to replace password in config file when systemd service starts.